### PR TITLE
[Toolchain] Input password in case root request

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit input user machine password using quick input
when there is a root request in command lists.

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

File: src/View/PasswordQuickInput.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  return await vscode.window.showInputBox({
+    title: 'Enter password',
+    password: true,
+    validateInput: validateInputIsAscii,
+  });
+
+  async function validateInputIsAscii(str: string) {
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    // NOTE ASCII characters have codes ranging from u+0000 to u+007f
+    return !/^[\u0000-\u007f]*$/.test(str) ? 'Password contains non-ASCII characters' : undefined;
```
Comment (hyunsik-yoon) [inline src/View/PasswordQuickInput.ts:29]: I was curious what `007f` was. I found that `DELETE` was `007f`. 
Question is 'is it OK with this validation rule (DELETE is not a char)'? I guess that we will ask Ubuntu `sudo` if this password is correct and `sudo` will check again. 
So I think this check is weak but there's no problem.

**Review by hyunsik-yoon — DISMISSED** [review]

File: src/Toolchain/ToolchainEnv.ts [hunk 2]
```diff
@@ -75,7 +76,21 @@ class Env implements BuilderJob {
...
+          console.log('Password dialog canceled');
+          return;
+        }
+        console.log('Enter the password: ' + password);
+        process.env.userp = password;
+        this.workFlow.start(this.currentWorkspace);
+      });
+    } else {
+      this.workFlow.start(this.currentWorkspace);
+    }
```
Comment (yunjayh) [inline src/Toolchain/ToolchainEnv.ts:93]: How about to remove printing logs to the console, at least printing password? I'm not sure that printing the user typed password is good to remain in main branch.

Comment (jyoungyun) [reply src/Toolchain/ToolchainEnv.ts:93]: Oh, it's my test code :) I will remove it. :)

Comment (jyoungyun) [reply src/Toolchain/ToolchainEnv.ts]: Other logs are came from vscode ssh extension.
You can see the same log before you enter the your remote machine password.

```
[12:52:49.543] > jyoung-linux@10.113.111.58's password: 
[12:52:49.544] Showing password prompt
[12:53:16.924] Got password response
[12:53:16.924] "install" wrote data to terminal: "********"
```

File: src/View/PasswordQuickInput.ts (hunk 1 above)
Comment (jyoungyun) [reply src/View/PasswordQuickInput.ts:29]: You can see the detail information about ASCII characters.
https://en.wikipedia.org/wiki/Basic_Latin_(Unicode_block)#:~:text=The%20block%20contains%20all%20the,alphabet%20and%20a%20control%20character.

another examples:
https://thewebdev.info/2022/01/25/how-to-check-if-a-string-has-any-non-ascii-characters-in-it-with-javascript/
https://stackoverflow.com/questions/14313183/javascript-regex-how-do-i-check-if-the-string-is-ascii-only

**Review by yunjayh — DISMISSED** [review]
LGTM. Thank you!!

File: src/Toolchain/ToolchainEnv.ts [hunk 3]
```diff
@@ -119,10 +134,13 @@ class ToolchainEnv extends Env {
 
   install(toolchain: Toolchain) {
     if (this.installed !== undefined) {
-      throw Error('Before installing, uninstall toolchain');
+      Balloon.error(
+          'Debian toolchain is allowed to install only one. Please uninstall toolchain first.');
```
Comment (dayo09) [inline src/Toolchain/ToolchainEnv.ts:138]:
```suggestion
          'The toolchain installed in your system must be one and only. Please uninstall the existing toolchain first.');
```
(optional) or one->once.

File: src/Toolchain/ToolchainEnv.ts [hunk 4]
```diff
@@ -119,10 +134,13 @@ class ToolchainEnv extends Env {
 
   install(toolchain: Toolchain) {
     if (this.installed !== undefined) {
-      throw Error('Before installing, uninstall toolchain');
+      Balloon.error(
+          'The toolchain installed in your system must be one and only. Please uninstall the existing toolchain first.');
```
Comment (jyoungyun) [inline src/Toolchain/ToolchainEnv.ts:138]:
```suggestion
          'The debian toolchain installed in your system must be one and only. Please uninstall the existing toolchain first.');
```

**Review by hyunsik-yoon — APPROVED** [review]

**Review by yunjayh — APPROVED** [review]

**Comment by jyoungyun** [comment]
PTAL @dayo09

**Review by dayo09 — APPROVED** [review]

**Pull request merged**